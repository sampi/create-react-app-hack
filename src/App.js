import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

// This is just a Thing
class Thing extends Component {
	render() {
		return <div className="thing" />;
	}
}

class App extends Component {
	render() {
		return (
			<div className="App">
				<header className="App-header">
					<img src={logo} className="App-logo" alt="logo" />
					<h1 className="App-title">Welcome to React</h1>
				</header>
				<p className="App-intro">
					To get started, edit <code>src/App.js</code> and save to reload.
				</p>
				<Thing />
			</div>
		);
	}
}

export default App;
