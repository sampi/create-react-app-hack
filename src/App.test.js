import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<App />, div);
	ReactDOM.unmountComponentAtNode(div);
});

it('can render into snapshots', () => {
	const tree = global.render(<App />);
	expect(tree).toMatchSnapshot();
});

it('can shallow render into snapshots', () => {
	const tree = global.shallowRender(<App />);
	expect(tree).toMatchSnapshot();
});
