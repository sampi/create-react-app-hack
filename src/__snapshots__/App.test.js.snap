// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render into snapshots 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <img
      alt="logo"
      class="App-logo"
      src="logo.svg"
    />
    <h1
      class="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    class="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <div
    class="thing"
  />
</div>
`;

exports[`can shallow render into snapshots 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <img
      alt="logo"
      class="App-logo"
      src="logo.svg"
    />
    <h1
      class="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    class="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <Thing />
</div>
`;
